### PR TITLE
add cask file for BitTorrent app 

### PR DESCRIPTION
**Pull request by daehyunpy**
add cask file: bittorrent.rb

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Please look at other casks to see how they’re structured. That matters.

File: Casks/bittorrent.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  url 'http://download-new.utorrent.com/os/osx/track/stable/endpoint/btmac'
+  name 'BitTorrent'
+  homepage 'http://www.bittorrent.com'
+  license :freemium
+
+  installer :manual => 'Bittorrent.app'
+  uninstall :trash =>
+    [
+      '/Applications/BitTorrent.app',
```
Comment (vitorgalvao) [inline Casks/bittorrent.rb]: Are you sure the only thing the installers does is place this app?

Comment (hyuna917) [inline Casks/bittorrent.rb]: hmm, i’m not sure. i just checked what i delete for uninstall by ‘cleanmymac 3’ app.

**Comment by hyuna917** [comment]
okay, give me a time to fix it.

File: Casks/bittorrent.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+cask :v1 => 'bittorrent' do
+  version :latest
+  sha256 :no_check
+
+  url 'http://download-new.utorrent.com/os/osx/track/stable/endpoint/btmac'
+  name 'BitTorrent'
+  homepage 'http://www.bittorrent.com'
+  license :freemium
```
Comment (vitorgalvao) [inline Casks/bittorrent.rb]: Is this a free app that works indefinitely but with limitations that can be removed by paying?

**Comment by vitorgalvao** [comment]
To be correctly structured, cask should be

```ruby
cask :v1 => 'bittorrent' do
  version :latest
  sha256 :no_check

  url 'http://download-new.utorrent.com/os/osx/track/stable/endpoint/btmac'
  name 'BitTorrent'
  homepage 'http://www.bittorrent.com'
  license :freemium

  installer :manual => 'Bittorrent.app'

  uninstall :trash => '/Applications/BitTorrent.app'

  zap :trash => [
                 '~/Library/Preferences/com.com.bittorrent.BitTorrent.plist',
                 '~/Library/Application Support/BitTorrent',
                 '~/Library/Saved Application State/com.com.bittorrent.BitTorrent.savedState',
                 '~/Library/Caches/com.com.bittorrent.BitTorrent',
                 '~/Library/LaunchAgents/com.bittorrent.BitTorrent.plist'
                ]
end
```

**Comment by vitorgalvao** [comment]
Don’t use `trash`, as it has not been implemented.

**Comment by hyuna917** [comment]
Was using `zap :trash` okay?

File: Casks/bittorrent.rb (hunk 2 above)
Comment (vitorgalvao) [inline Casks/bittorrent.rb]: This still needs to be clarified.

File: Casks/bittorrent.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  installer :manual => 'Bittorrent.app'
+
+  uninstall :delete => '/Applications/BitTorrent.app'
```
Comment (vitorgalvao) [inline Casks/bittorrent.rb]: Have you confirmed this is the only thing it does?

**Comment by vitorgalvao** [comment]
`trash` is currently just an alias for `delete`, so it should not be used.

File: Casks/bittorrent.rb (hunk 2 above)
Comment (hyuna917) [inline Casks/bittorrent.rb]: yes, we can use it indefinitely. but, just limited function.

Comment (hyuna917) [inline Casks/bittorrent.rb]: do you think it’s not ‘freemium’ but ‘gratis’? hmm, maybe it is. i’m not sure i understand the difference.

**Comment by vitorgalvao** [comment]
What are the limitations it has?

**Comment by hyuna917** [comment]
just add-free, built-in player, anti-virus, etc⋯ 

http://www.bittorrent.com/bittorrent-pro

**Comment by vitorgalvao** [comment]
Seems like Pro is Windows-only.

**Comment by vitorgalvao** [comment]
Thank you for the contribution. It needed some fixes, so they were made in 22b4adba8e075c259bc15c969067e037c273bd89. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

**Pull request closed without merging**